R/getTimeDivergence.R

```diff
@@ -58,21 +60,24 @@ getTimeDivergence <- function(se, sample_field, sample_id, time_field, time_inte
     mat[, time_field][location] <- extracted_time
 
     #related samples are chosen from assay
-    x <- as.matrix(assay(tse)[,rownames(sample_df)[which(is.na(mat[, time_field]) == FALSE)]])
+    x <- as.matrix(assay(se)[,rownames(sample_df)[which(is.na(mat[, time_field]) == FALSE)]])
 
     if(!transposed){
       x <- t(x)
     }
 
-    timedivergence_n <- as.matrix(vegan::vegdist(x), "bray")
+    #timedivergence_n <- calculateDistance(x)
+    timedivergence_n <- calculateJSD(x)
 
-    #timedivergence_n <-do.call(stats::dist, list(x))
-    #timedivergence_n <- as.data.frame(as.matrix(time_divergence_n))
+    mat.m <- matrixConvert(timedivergence_n, colname = c("sp1", "sp2", "dist"))
 
-    time_diff <- diff(extracted_time)
+    mat.m$timedifference_n <- vapply(seq_len(length(timedivergence_n)),
+                                     FUN = function(i) {extracted_time[mat.m$sp2[i]] - extracted_time[mat.m$sp1[i]]},
+                                     FUN.VALUE = 1)
 
-    timedifference_n <- c(NA, time_diff)
 
-    return(list(timedivergence_n = timedivergence_n,
-                timedifference_n = timedifference_n))
+    mat.m$sample_id <- c(sample_id)
+    mat.m <- mat.m[,c("sample_id", "sp1", "sp2", "dist", "timedifference_n")]
+
+    return(mat.m)
```

Comment:
Return is a matrix containing the samples used in the beta diversity and the time difference between them. I am still not sure how to implement it back to colData since each row has 2 samples that are being considered for beta diversity calculation. The only way I can think is to have separate colData field from the original one.

Reply:
There is no need to return sample profiles to colData. The colData should have 1) beta diversity i.e. difference between sample T and sample T-t; 2) time difference between time steps for sample T-t. Both of these are univariate.

Reply:
Here T meaning the latter and t meaning the former sample. By default, consecutive time steps but we also allow longer intervals (differences between two time steps or more, optionally).

Reply:
Sample profiles we already have in the abundance table (assay), not sure why they would be needed in colData.

Reply:
Then the comment about colData in #13 is meant different from what I understood. 

Time difference between T-t is there. For 1) beta diversity between sample T and sample t is there too, but that's not what you meant, I guess. Okay, let me give an example. For instance, here in this example, Baboon-1 has 6 different time steps. In the case of time interval = 2, only the samples at t = 2, t = 4, t= 6 used for beta calculation. So the beta calculation is between t2-t4, t2-t6 and t4-t6 and the time difference between these also calculated. And sp1 and sp2 fields in return matrix show the samples used in the beta diversity.

Reply:
With time interval = 2, we need the beta diversities for t2/t4, t3/t5, t4/t6 i.e. all pairs that are separated by 2 time steps. The pair t2-t6 in your example above is separated by 4 time steps.